src/jshint.js

```diff
@@ -5071,12 +4993,16 @@ var JSHINT = (function() {
       },
 
       current: {
+        getlabel: function(t) {
```

Comment:
> but I think this method might be too powerful. As defined, the previously-internal "label" objects are exposed, and we'll have to be vigilant about how future patches handle those.

If you look above, you'll see that there is another `getlabel` function already that returns a label object and there is code in multiple places calling it and also changing the returned object.
The only difference with the new function is it only looks in the current scope, where as the existing function looks in all blockscopes 

(see this code)
```
block = funct["(blockscope)"].getlabel(v);
```

but you are right that a labeltype would reduce the repetition - I will change this to labeltype and add a labeltype in the non-`current` context and use that as well.